esp8266-fastled-webserver/Fields.cpp

```diff
@@ -17,479 +17,526 @@
+      {"brightThetaIncMin",    "Bright Theta Inc Min",   Field_t::Number,   0, 255, getBrightThetaIncMin,    nullptr, setBrightThetaIncMin},
+      {"brightThetaIncMax",    "Bright Theta Inc Max",   Field_t::Number,   0, 255, getBrightThetaIncMax,    nullptr, setBrightThetaIncMax},
+      {"msMultiplierBpm",      "Time Multiplier BPM",    Field_t::Number,   0, 255, getMsMultiplierBpm,      nullptr, setMsMultiplierBpm  },
+      {"msMultiplierMin",      "Time Multiplier Min",    Field_t::Number,   0, 255, getMsMultiplierMin,      nullptr, setMsMultiplierMin  },
+      {"msMultiplierMax",      "Time Multiplier Max",    Field_t::Number,   0, 255, getMsMultiplierMax,      nullptr, setMsMultiplierMax  },
+      {"hueIncBpm",            "Hue Inc BPM",            Field_t::Number,   0, 255, getHueIncBpm,            nullptr, setHueIncBpm        },
+      {"hueIncMin",            "Hue Inc Min",            Field_t::Number,   0, 255, getHueIncMin,            nullptr, setHueIncMin        },
+      {"hueIncMax",            "Hue Inc Max",            Field_t::Number,   0, 255, getHueIncMax,            nullptr, setHueIncMax        },
+      {"sHueBpm",              "S Hue BPM",              Field_t::Number,   0, 255, getSHueBpm,              nullptr, setSHueBpm          },
+      {"sHueMin",              "S Hue Min",              Field_t::Number,   0, 255, getSHueMin,              nullptr, setSHueMin          },
+      {"sHueMax",              "S Hue Max",              Field_t::Number,   0, 255, getSHueMax,              nullptr, setSHueMax          },
+  };
+
+  Field getField(String name) {
+    return getField(name, fields);
   }
 
-  json += "]";
+} // end inline anonymous namespace
+
+// Everything else is exported from this file....
 
-  return json;
+void setShowClock(uint8_t value)
+{
+  showClock = value == 0 ? 0 : 1;
+  writeAndCommitSettings();
+  broadcastInt("showClock", showClock);
 }
+void setClockBackgroundFade(uint8_t value)
+{
+  clockBackgroundFade = value;
+  writeAndCommitSettings();
+  broadcastInt("clockBackgroundFade", clockBackgroundFade);
+}
+uint8_t setUtcOffsetIndex(uint8_t value) {
+  utcOffsetIndex = value;
+  if (utcOffsetIndex > 104) utcOffsetIndex = 104; // 0 === -12 hours, 104 === +14 hours
+  
+  const int32_t UTC_OFFSET_MINIMUM_MINUTES = ((int32_t)-12) * 60; // corresponds to index 0
+  const int32_t UTC_OFFSET_INCREMENT_MINUTES = 15; // each higher index increments by this amount
+  
+  // minutes above the minimum
+  int32_t tmp = utcOffsetIndex * UTC_OFFSET_INCREMENT_MINUTES;
+  // add that to the minimum value
+  tmp = UTC_OFFSET_MINIMUM_MINUTES + tmp;
+  // convert to seconds
+  utcOffsetInSeconds = tmp * 60;
+
+  Serial.print(F("utcOffsetIndex: ")); Serial.println(utcOffsetIndex);
+  Serial.print(F("utcOffsetInSeconds: ")); Serial.println(utcOffsetInSeconds);
+  timeClient.setTimeOffset(utcOffsetInSeconds);
+  writeAndCommitSettings();
+  return utcOffsetIndex;
+}
+
```

Comment:
I do believe this should not go into the Fields.hpp file but somewhere else. The `writeAndCommitSetting()` is currently only used in the main "sketch"...

Reply:
Yes, I agree.  In fact, I would argue that all the get/set functions should be moved outside this file, as well as the variables themselves.

Would you open an issue to move the getter/setter functions into their own file?

Reply:
Now tracked in #220.